dask_expr/_expr.py

```diff
@@ -2758,18 +2764,29 @@ def normalize_expression(expr):
     return expr._name
 
 
-def optimize_until(expr: Expr, stage: core.OptimizerStage) -> Expr:
+def optimize_until(
+    expr: Expr, stage: core.OptimizerStage, common_subplan_elimination: bool = False
+) -> Expr:
     result = expr
     if stage == "logical":
         return result
 
-    # Simplify
-    expr = result.simplify()
+    while True:
+        if not common_subplan_elimination:
+            out = result.rewrite("reuse", cache={})
```

Comment:
While we have cleaned up the meaning of `common_subplan_elimination`, it's still weird to me that we execute the `reuse` rule to _avoid_ CSE. Maybe we should rename it to something like `avoid_common_subplan_elimination`?